models/shift_net/shiftnet_model.py

```diff
@@ -273,13 +273,14 @@ def backward_G(self):
                                + self.criterionGAN (self.pred_fake - torch.mean(self.pred_real), True)) / 2.
 
 
-        self.loss_G_L1 = 0
+        self.loss_G_L1, self.loss_G_L1_m = 0, 0
         self.loss_G_L1 += self.criterionL1(self.fake_B, self.real_B) * self.opt.lambda_A
+        self.loss_G_L1_m += self.criterionL1(self.fake_B*self.mask_global.float(), self.real_B*self.mask_global.float())*self.opt.mask_weight
 
         if self.wgan_gp:
-            self.loss_G = self.loss_G_L1 - self.loss_G_GAN * self.opt.gan_weight
+            self.loss_G = self.loss_G_L1 + self.loss_G_L1_m - self.loss_G_GAN * self.opt.gan_weight
         else:
-            self.loss_G = self.loss_G_L1 + self.loss_G_GAN * self.opt.gan_weight
+            self.loss_G = self.loss_G_L1 + self.loss_G_L1_m + self.loss_G_GAN * self.opt.gan_weight
```

Comment:
Maybe using `Discounted_L1` from contextual attention?

Reply:
We could :)